stablehlo/reference/Element.cpp

```diff
@@ -538,6 +564,38 @@ Element min(const Element &e1, const Element &e2) {
       });
 }
 
+Element power(const Element &e1, const Element &e2) {
+  Type type = e1.getType();
+
+  if (isSupportedIntegerType(type)) {
+    APInt base = e1.getIntegerValue();
+    APInt exponent = e2.getIntegerValue();
+    bool isSigned = base.isSignedIntN(base.getBitWidth());
+    if (isSigned && exponent.isNegative()) {
+      if (base.isZero())
+        llvm::report_fatal_error("divide by zero unsupported");
+      else if (base.abs().isOne())
+        exponent = exponent.abs();
+      else
+        return Element(type, APInt(base.getBitWidth(), 0, isSigned));
```

Comment:
`return Element(type, 0)`?

Reply:
Apologies - I'm realizing that the proposed code doesn't compile. You'll want to say `return Element(type, 0l)` to disambiguate.